api/src/pcapi/routes/backoffice/finance/finance_incidents_blueprint.py

```diff
@@ -881,6 +883,222 @@ def get_finance_incident_validation_form(finance_incident_id: int) -> utils.Back
+    for incident in finance_incidents:
+        with atomic():
+            try:
+                finance_api.cancel_finance_incident(
+                    incident,
+                    comment=form.comment.data,
+                    author=current_user,
+                )
+                success_count += 1
+            except (
+                finance_exceptions.FinanceIncidentAlreadyCancelled,
+                finance_exceptions.FinanceIncidentAlreadyValidated,
+            ) as err:
+                error_dict[err.__class__.__name__].append(incident.id)
+                mark_transaction_as_invalid()
+
+    _flash_success_and_error_messages(success_count, error_dict, is_validating=False)
+    return redirect(request.referrer or url_for("backoffice_web.finance_incidents.list_finance_incidents"), 303)
+
+
+def _build_incident_error_str(incident_ids: list[str], message: str) -> str:
+    return Markup("- {count} {message} pour {incident} ({incident_ids})").format(
+        count=len(incident_ids),
+        message=message,
+        incident=pluralize(len(incident_ids), "l'incident", "les incidents"),
+        incident_ids=",".join(incident_ids),
+    )
+
+
+def _flash_success_and_error_messages(
+    success_count: int, error_dict: dict[str, list[str]], is_validating: bool
+) -> None:
+    if success_count > 0:
+        flash(
+            f"{success_count} {pluralize(success_count, 'incident a', 'incidents ont')} été {'validé' if is_validating else 'annulé'}{pluralize(success_count)}",
+            "success",
+        )
+
+    if error_dict:
+        if success_count > 0:
+            error_text = Markup("Certains incidents n'ont pas pu être {action} et ont été ignorés :<br> ").format(
+                action="validés" if is_validating else "annulés"
+            )
+        else:
+            error_text = Markup("Impossible {action} ces incidents : <br>").format(
+                action="de valider" if is_validating else "d'annuler"
+            )
+        for key in error_dict:
+            incident_ids = [str(incident_id) for incident_id in error_dict[key]]
+            match key:
+                case "FinanceIncidentAlreadyCancelled":
+                    error_text += _build_incident_error_str(
+                        incident_ids,
+                        f"incident{pluralize(len(incident_ids))} déjà annulé{pluralize(len(incident_ids))}",
+                    )
+                case "FinanceIncidentAlreadyValidated":
+                    error_text += _build_incident_error_str(
+                        incident_ids,
+                        f"incident{pluralize(len(incident_ids))} déjà validé{pluralize(len(incident_ids))}",
+                    )
+                case _:
+                    error_text += _build_incident_error_str(
+                        incident_ids,
+                        f"""Erreur{pluralize(len(incident_ids))} {pluralize(len(incident_ids), "s'est produite", "se sont produites")} : "{key}" """,
```

Comment:
Détail : c'est le nombre qui commence la phrase.
```suggestion
                        f"""erreur{pluralize(len(incident_ids))} {pluralize(len(incident_ids), "s'est produite", "se sont produites")} : "{key}" """,
```